docs/testing-tool.rst

```diff
@@ -7,85 +7,118 @@ Use the KSQL testing tool to test a set of KSQL statements. The KSQL testing too
+
+
+            --sql-file <statementsFile>, -s <statementsFile>
+                A SQL file containing KSQL statements to be tested.
+
+                This option may occur a maximum of 1 time
+
 
 
 Test File Structure
 *******************
-The test file is a JSON file containing the KSQL statements, input data, desired configurations, and the expected results or expected errors.
-The following is a sample test file:
+
+Statements File
+---------------
+
+The statements file contains the KSQL statements to test. The following are the supported statements in the testing tool:
+
+- CREATE STREAM
+- CREATE TABLE
+- CREATE STREAM AS SELECT
+- CREATE TABLE AS SELECT
+- INSERT INTO SELECT
+
+Here is a sample statements file for the testing tool:
+
+.. code:: sql
+    CREATE STREAM SRTREAM1 (ID bigint, NAME varchar, VALUE double) WITH (kafka_topic='test_topic', value_format='DELIMITED', key='ID');
+    CREATE STREAM SRTREAM2 as SELECT name FROM SRTREAM1 WHERE id > 100;
+
+Input File
+----------
+
+The input file is a JSON file with one array field named "inputs". Each element in the array is the representation of input messages.
+A message should have a topic, a key, a value and a timestamp. The following is a sample input file for the above test:
 
 .. code:: json
     {
-      "comments": [
-        "Add a description of the functionality that this file tests"
-      ],
-      "tests": [
-        {
-          "name": "my first positive test",
-          "description": "an example positive test where the output is verified",
-          "statements": [
-            "CREATE STREAM intput (ID bigint) WITH (kafka_topic='input_topic', value_format='JSON');",
-            "CREATE STREAM output AS SELECT id FROM intput WHERE id < 10;"
-          ],
-          "inputs": [
-            {"topic": "input_topic", "key": 0, "value": {"id": 8}, "timestamp": 0},
-            {"topic": "input_topic", "key": 0, "value": {"id": 10}, "timestamp": 10000},
-            {"topic": "input_topic", "key": 1, "value": {"id": 9}, "timestamp": 30000},
-            {"topic": "input_topic", "key": 1, "value": {"id": 11}, "timestamp": 40000}
-          ],
-          "outputs": [
-            {"topic": "OUTPUT", "key": 0, "value": {"ID": 8}, "timestamp": 0},
-            {"topic": "OUTPUT", "key": 1, "value": {"ID": 9}, "timestamp": 30000}
-          ]
-        },
-        {
-          "name": "my first negative test",
-          "description": "an example negative test where the statement will fail to parse",
-          "statements": [
-            "CREATE STREAM TEST WITH (kafka_topic='test_topic', value_format='DELIMITED');"
-          ],
-          "expectedException": {
-            "type": "io.confluent.ksql.util.KsqlException",
-            "message": "The statement does not define any columns."
-          }
-        }
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": "0,zero,0.0"},
+        {"topic": "test_topic", "key": 100, "value": "100,100,0.0"},
```

Comment:
nit: I was confused by this example at first since at first glance it looks like the types don't match. Turns out it works because the mismatched column is of type `VARCHAR`, but it'd probably be better to update the example to avoid this confusion.

Reply:
Yes, since it's Delimited format it takes care of casting. I changed the example to JSON so it would remove any confusions!